src/handle_commit.cxx

```diff
@@ -469,12 +470,30 @@ bool raft_server::snapshot_and_compact(ulong committed_idx, bool forced_creation
 
     if (!forced_creation) {
         // If `forced_creation == true`, ignore below conditions.
+        auto snapshot_distance = (ulong)params->snapshot_distance_;
+
+        // Randomized snapshot distance for the first creation.
+        if ( params->snapshot_distance_ != 0
+            && !last_snapshot_
+            && params->enable_randomized_snapshot_creation_ ) {
+            uint seed = (uint)( std::chrono::system_clock::now()
+                                   .time_since_epoch().count() * id_ );
+            std::default_random_engine engine(seed);
+            std::uniform_int_distribution<int32>
+                distribution( 0, params->snapshot_distance_ / 2 );
+
+            snapshot_distance = params->snapshot_distance_ / 2
+                                + distribution(engine);
+
+            p_in("First snapshot creation log distance %llu", snapshot_distance);
```

Comment:
It will not provide enough randomness, as mostly all members will hit this logic at the same time.

I recommend defining a member variable
```c++
    /**
     * (Read-only)
     * The first snapshot distance.
     */
    uint32_t first_snapshot_distance_;
```
and initialize it in the constructor of `raft_server` like this, based on the timestamp of Raft server initialization:
```c++
, first_snapshot_distance_(ctx_->get_params()->snapshot_distance_)
...

    ptr<raft_params> params = ctx_->get_params();
    if (params->stale_log_gap_ < params->fresh_log_gap_) {
        params->stale_log_gap_ = params->fresh_log_gap_;
    }
    if (params->enable_randomized_snapshot_creation_) {
        uint64_t seed = timer_helper::get_timeofday_us() * id_;

        // Flip the integer.
        uint8_t* first = reinterpret_cast<uint8_t*>(&seed);
        uint8_t* last = reinterpret_cast<uint8_t*>(&seed) + sizeof(seed);
        while ((first != last) && (first != -- last)) std::swap(*first++, *last);

        std::default_random_engine engine(seed);
        std::uniform_int_distribution<int32>
            distribution( params->snapshot_distance_ / 2, params->snapshot_distance_ );

        first_snapshot_distance_ = distribution(engine);
        p_in("First snapshot creation log distance %llu", first_snapshot_distance_);
    }
```

And this if-clause just does
```c++
snapshot_distance = first_snapshot_distance_;
```